vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractRow.java

```diff
@@ -355,8 +355,11 @@ private CELL joinCellsInPlace(Collection<CELL> cellsToJoin,
         layer.addColumn(cellInsertIndex, group);
 
         layer.getColumns().removeAll(columnsToJoin);
-
-        this.cells.removeAll(cellsToJoin);
+        CELL keeper = this.cells.get(elementInsertIndex);
+        this.cells.removeAll(cellsToJoin
```

Comment:
I dont understand this change, why do you filter and collect if the returned result is not used anywhere, or am I wrong?